Update the Update and Bulk API's  update/upsert sections (#6182)
opensearch-project/documentation-website: _api-reference/document-apis/update-document.md

**Introduced 1.0**
{: .label .label-purple }

> If you need to update a document's fields in your index, you can use the update document API operation. You can do so by specifying the new data you want in your index or by including a script in your request body, which OpenSearch runs to update the document. By default, the update operation only updates a document that exists in the index. If a document does not exist, the API returns an error. To _upsert_ a document (update the document that exists or index a new one), use the [upsert](#upsert) operation.

Reviewer: "you want to be in"?

## Request body

> Your request body must contain the information you want to update your document with. If you only want to replace certain fields in your document, your request body must include a `doc` object, which has the fields you want to update:

Reviewer: ```suggestion
Your request body must contain the information with which you want to update your document. If you only want to replace certain fields in your document, your request body must include a `doc` object containing the fields that you want to update:
```

}
```

## Upsert

Upsert is an operation that conditionally either updates an existing document or inserts a new one based on information in the object. 

> In the following example, the `upsert` operation updates the `last name` and adds the `first_name` field if a document already exists. If a document does not exist, a new one is indexed using content in the `upsert` object.

Reviewer: ```suggestion
In the following example, the `upsert` operation updates the `last name` field and adds the `first_name` field if a document already exists. If a document does not exist, a new one is indexed using content in the `upsert` object.
```

Author: Reworded.

  }
}
```

Consider an index that contains the following document:

```json
{
  "_index": "sample-index1",
  "_id": "1",
  "_score": 1,
  "_source": {
    "first_name": "Bruce",
    "last_name": "Wayne"
  }
}
```

After the upsert operation, the document's `first_name` and `last_name` fields are updated:

```json
{
  "_index": "sample-index1",
  "_id": "1",
  "_score": 1,
  "_source": {
    "first_name": "Martha",
    "last_name": "Rivera"
  }
}
```

If the document does not exist in the index, a new document is indexed with the fields specified in the `upsert` object:

```json
{
  "_index": "sample-index1",
  "_id": "1",
  "_score": 1,
  "_source": {
    "last_name": "Oliveira",
    "age": "31"
  }
}
```

> You can also add `doc_as_upsert` to the request and set it to `true` to use the information in `doc` for performing the upsert operation:

Reviewer: "in the `doc` field"?

> You can also add `doc_as_upsert` to the request and set it to `true` to use the information in `doc` for performing the upsert operation:

Author: ```suggestion
You can also add `doc_as_upsert` to the request and set it to `true` to use the information in the `doc` field for performing the upsert operation:
```
